docs/proposals/PROPOSAL_H.md

```diff
@@ -108,103 +110,172 @@ Schema:
 
-If the field is required in image compatibility artifact, but does not present in NFD labels, the new feature lable should be discussed and contributed into NFD.
+    - **`architecture`** _string_
 
-It's expected that NFD feature lables definition and node feature discovering could be an independent component in NFD, and could be consumed by different consumers.
+    - **`variant`** _string_
 
-To avoid over-flexible specification, and avoid abusing the specification, a way to limit the freedom is needed.
+    - **`os`** _string_
+
+    - **`os.version`** _string_
+
+    - **`os.features`** _array of strings_
+
+    - **`features`** _array of strings_
+
+    - **`compatibilities`** _list of map_
+
+      This new OPTIONAL property describes a list of compatibility sets, contains at least one compatibility set. Same definition as that in Image Compatibility Spec.
+
+### Migration and Deprecation
 
-## Artifact Discovery
+The platform.os.version, platform.os.features and platform.features are expected to be migrated to platform.compatibilites fields.
+
+It may take years for the migration.
+
+The deprecation for these fields will start with pre-notification and migration transition period.
+
+Before all runtimes' migration finished, these fields should still be valid.
+
+## Flexibility for Different Purpose
+
+It's the image author's (or image provider's) freedom to use standalone artifact or platform.compatibilities to express the compatibility requirements, the format will be kept same.
 
 The subject field should be used to associate the compatibility artifact with target image. The referrers API should be used to discover artifacts.
 
 The client will query for an artifact using the referrers response, selecting the entry with a matching artifactType and the most recent "created" timestamp.
 
+If platform.compatibilities is detected and not empty, then runtime is not required to send referrs API to query compatibility artifact.
+
+If there is no platform.compatibilities, or the content of platform.compatibilities is  empty, then runtime should send referrs API to registry to query compatibility artifact.
+
+## Pros. and Cons. Comparation
+
+### Pros. and Cons. of Compatibility Artifact
+
+It's possible to update compatibility independently without having to re-release and re-distribute image.
+
+In some vertical industries, re-distributing image means repeating release cycle to production site, it may take several weeks to several months for complex applications, very time consuming and high cost.
+
+The application using new image has to pass series tests inside vendor's lab and then customer's lab, and then roll out in production site gradually. Tests may include function, performance, reliability, security, inter-working etc.
+
+The cons. of compatibility artifact is, if it's used for image selection or scheduling decision, additional referrs API request should be sent to registry.
```

Comment:
This raises two questions for me. Does a company need to retest an image if the individual platform digests do not change, only the image index is updated? And conversely, if the compatibility artifact is changed that modifies how runtimes would select an image to deploy in production, should that be excluded from any testing?

Reply:
Good questions.

> Does a company need to retest an image if the individual platform digests do not change, only the image index is updated?

In software package(or image package) delivery mode, the files are transfered from the vendor to the customer, if a package for example (tar.gz file) is updated, test in customer environment is needed. 

it's possible to update image index(for example, platform.compatibilities) in devops environment independently, because the creator and production maintainence engineer are consided in a same responsibility team. but in software package delivery mode, the creator(software supplier) and the customer(the buyer) have different responsibility boundary, if the customer can manupulate the image index on his own, it may lead to production environment out of control.

> And conversely, if the compatibility artifact is changed that modifies how runtimes would select an image to deploy in production, should that be excluded from any testing?

Can't be excluded from all tests, but can save some tests. If compatibility artifact is changed, the comaptibility artifact validation for production or lab testbed is needed too. If the image unchanged, and existing production environment is also same, and the new released compatibility artifact is validated compatible to existing production environment, then this kind of test can be saved. That means we don't have to upgrade the production image, just let it run as past.

If some new hardware or new os system is introduced(updated) in the compatibility artifact, those tests for the new hardware and new os system (which still run the existing images) can't be excluded from R&D lab to customer lab and production environment.

one vendor may have 100 customers, 90 customers may still use old production hardware and os environment, then update the artifact will not push these customers to do image upgrade in production, but other 10 customers which may use some new hardwares or os environments, then tests in these 10 customers are inevitable.

Reply:
> if the customer can manupulate the image index on his own, it may lead to production environment out of control.

I believe this same risk exists when using a compatibility artifact, but it is somewhat obscured from the end user because the index digest isn't changing. Just as a change to the index with an identical collection of image digests should receive an abbreviated review, so should any change to the compatibility artifact. Treating one change as different from another I worry is either a false sense of security or a false fear of what has changed.

Reply:
Your concerns make sense(currently I haven't find tool to directly update platform fields). It's up to how the vendor deliver package, and what's the review/modification process the customer will adopt to apply a package delivered by the vendor.

Reply:
Currently artifact can be a standalone deliverable. That means it can be delivered to customer as an independent package. I am not sure whether the platform.compatibilities can be a standalone deliverable, it's reasonable the fields in platform should be part of image deliverable.

Reply:
@sudo-bmitch maybe these two comments  are not good enough. https://github.com/opencontainers/wg-image-compatibility/pull/17#discussion_r1565130511 and https://github.com/opencontainers/wg-image-compatibility/pull/17#discussion_r1568090059 . 

do you think that the proposal need to include update like: from security level, both artifact and index update should be reviewed and processed in identical/similar way.